docs/databases/connecting.md

```diff
@@ -45,6 +45,70 @@ For provider-specific connection details, like connecting to a PostgreSQL data w
 
 - [AWS's Relational Database Service (RDS)](./connections/aws-rds.md)
 
+## Database roles, users, and privileges
+
+For Metabase to connect, query, or write to your database, you must give Metabase a database user account with the correct database privileges.
+
+The easiest way to set this up in a Postgres database:
+
+- Create a `metabase` database user.
+- Give `metabase` read and write access to the entire database.
+
+```sql
+CREATE USER metabase;
+GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA "your_schema" TO metabase;
+```
+
+You can always revoke specific privileges later. If you'd prefer to set things up with a little more structure, see the next couple of sections on:
+
+- [Database roles and users](#database-roles-and-users)
+- [Database privileges](#database-privileges)
+
+### Database roles and users
+
+To organize your database privileges, you can create database roles for each type of application that needs access to your database.
+
+For example, most BI tools need the same privileges to connect and read from your database, so you can:
+
+- Create a database role called `bi_tool`.
+- Create a database user called `metabase`.
+- Give `metabase` the `bi_tool` role.
+
+In PostgreSQL, you'd log in as an admin and run the SQL statements:
+
+```sql
+CREATE ROLE bi_tool;
+CREATE USER metabase;
+GRANT bi_tool TO metabase;
+```
+
+> If you're setting up multi-tenant permissions for people who need SQL access to specific schemas or tables, you may want to create [multiple database users](https://www.metabase.com/learn/permissions/multi-tenant-permissions#option-2-granting-customers-native-sql-access-to-their-schema) for Metabase, such as `metabase_customer_1` and `metabase_customer_2`.
+
+### Database privileges
+
+Once you've set up your [database roles and users](#database-roles-and-users), you can assign database privileges to those roles. When you grant privileges to a role, all users with that role will get those privileges.
+
+At minimum, the `bi_tool` role should be able to connect to and query your database:
+
+```sql
+GRANT CONNECT ON DATABASE "your_database" TO bi_tool;
```

Comment:
I like this second pass, but think we could go further. We should give people more complete code here, so people don't have to figure this out for themselves and can just copy/paste/modify.

In fact, we could/should recommend a cookie cutter set of SQL and roles to set up for Metabase. Like: tell them to create specific roles that have different privileges required for different Metabase features.

Something like this pseudocode:

```
-- Set up db user account that Metabase will use to connect to the database.
CREATE USER metabase;

-- Create roles with different privileges
--
-- Role with the minimum privileges for Metabase to work
CREATE ROLE connect_and_query;
GRANT CONNECT ON DATABASE "your_database" TO connect_and_query;
...

-- Role for Metabase actions and model caching to work
-- (Nat, I'm guessing on this here, we may need to split model caching and action privileges into separate roles)
CREATE ROLE actions_and_caching
GRANT CONNECT ...

-- Assign role(s) to `metabase` user account.
GRANT connect_and_query to metabase;
GRANT actions_and_caching to metabase;
```

And then, like you do below, also show how to do it for particular schema/tables.